### ISSUE TITLE
Issues with cpu percentage in host stats collections blocks host registration

### ISSUE THREAD

**Issue by ibuildthecloud** [rancherio/python-agent#74]
Two issues were seen:

First 
```python
Traceback (most recent call last):
  File "/var/lib/cattle/pyagent/cattle/agent/event.py", line 91, in _worker_main
    resp = agent.execute(req)
  File "/var/lib/cattle/pyagent/cattle/agent/__init__.py", line 15, in execute
    return self._router.route(req)
  File "/var/lib/cattle/pyagent/cattle/plugins/core/event_router.py", line 13, in route
    resp = handler.execute(req)
  File "/var/lib/cattle/pyagent/cattle/plugins/core/event_handlers.py", line 32, in execute
    type.on_ping(event, resp)
  File "/var/lib/cattle/pyagent/cattle/plugins/docker/compute.py", line 54, in on_ping
    self._add_resources(ping, pong)
  File "/var/lib/cattle/pyagent/cattle/plugins/docker/compute.py", line 92, in _add_resources
    'physicalHostStats': self.host_info.collect_data()
  File "/var/lib/cattle/pyagent/cattle/plugins/host_info/main.py", line 17, in collect_data
    data[collector.key_name()] = collector.get_data()
  File "/var/lib/cattle/pyagent/cattle/plugins/host_info/cpu.py", line 81, in get_data
    data.update(self._get_cpu_percentages())
  File "/var/lib/cattle/pyagent/cattle/plugins/host_info/cpu.py", line 52, in _get_cpu_percentages
    stat_prev['timestamp'])
  File "/var/lib/cattle/pyagent/cattle/utils.py", line 96, in timestamp_diff
    time_current_conv = self._timestamp_convert(time_current)
  File "/var/lib/cattle/pyagent/cattle/utils.py", line 105, in _timestamp_convert
    return datetime.strptime(t_strip, "%Y-%m-%dT%H:%M:%S.%f")
  File "/usr/lib/python2.7/_strptime.py", line 328, in _strptime
    data_string[found.end():])
ValueError: unconverted data remains: 87-0
```

Second
```
Traceback (most recent call last):
  File "/var/lib/cattle/pyagent/cattle/agent/event.py", line 91, in _worker_main
    resp = agent.execute(req)
  File "/var/lib/cattle/pyagent/cattle/agent/__init__.py", line 15, in execute
    return self._router.route(req)
  File "/var/lib/cattle/pyagent/cattle/plugins/core/event_router.py", line 13, in route
    resp = handler.execute(req)
  File "/var/lib/cattle/pyagent/cattle/plugins/core/event_handlers.py", line 32, in execute
    type.on_ping(event, resp)
  File "/var/lib/cattle/pyagent/cattle/plugins/docker/compute.py", line 54, in on_ping
    self._add_resources(ping, pong)
  File "/var/lib/cattle/pyagent/cattle/plugins/docker/compute.py", line 92, in _add_resources
    'physicalHostStats': self.host_info.collect_data()
  File "/var/lib/cattle/pyagent/cattle/plugins/host_info/main.py", line 17, in collect_data
    data[collector.key_name()] = collector.get_data()
  File "/var/lib/cattle/pyagent/cattle/plugins/host_info/cpu.py", line 81, in get_data
    data.update(self._get_cpu_percentages())
  File "/var/lib/cattle/pyagent/cattle/plugins/host_info/cpu.py", line 47, in _get_cpu_percentages
    stat_prev = stats[-2]
```

It seems the date format is not what is expected.  The dates coming back look like `2015-02-04T23:21:38.251266323-07:00`

Because of these issues the agent never successfully registers.

**Comment by cloudnautique** [comment]
What OS?

**Comment by ibuildthecloud** [comment]
Ubuntu 14.10, docker 1.5.0-rc3 if that matters

**Comment by cloudnautique** [comment]
This was addressed in pull #75
